File: TaskMaster_API/TaskMaster/Model/Domain/ProjectData/TestCase.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

using TaskMaster.Model.API.ProjectData;
using TaskMaster.Model.Domain.UserData;

namespace TaskMaster.Model.Domain.ProjectData;

public class TestCase
{
    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public int TestCaseId { get; set; }

    [Required]
    public string Title { get; set; }

    [Required]
    public string Details { get; set; }

    [Required]
    [ForeignKey("User")]
    public int UserId { get; set; }

    [Required]
    [ForeignKey("Project")]
    public int ProjectId { get; set; }

    public Project Project { get; set; }
    public User User { get; set; }

    
}

File: App/TaskMaster/TaskMaster/Models/TestCase.swift
//
//  TestCase.swift
//  TaskMaster
//
//  Created by Joshua Root on 12/4/23.
//

import SwiftData
import Foundation

@Model
class TestCase {
    var title: String
    var details: String
    var project: Project
    
    init(title: String, details: String, project: Project) {
        self.testCaseId = testCaseId
        self.title = title
        self.details = details
        self.project = project
    }

    
}
